### ISSUE TITLE
Integration tests: log: exiting because of error: log: unable to create log: open /tmp/minikube_<command>.log: permission denied

### ISSUE THREAD

**Issue by ilya-zuyev** [kubernetes/minikube#11242]
We have a flake failure in `TestFunctional/parallel/ProfileCmd/*` tests with output:

``` 13:28:05     functional_test.go:1077: (dbg) Non-zero exit: out/minikube-linux-amd64 profile list -o json: exit status 2 (60.447216ms)
13:28:05         
13:28:05         ** stderr ** 
13:28:05         	I0430 20:28:05.834451   78637 out.go:278] Setting OutFile to fd 1 ...
13:28:05         	log: exiting because of error: log: unable to create log: open /tmp/minikube_profile_cf3359060b829bcde9ec55fd26dfb85c50e69ef0_0.log: permission denied
13:28:05         
13:28:05         ** /stderr ** 
```

Could be a jenkins agent vm setup issue, or this file is accessed from multiple processes at the same time

**Comment by afbjorklund** [comment]
We have see other problems with the naming of the log files, when running multiple process or multiple users: #11205

So I think it's a real problem, not just a testing issue ?

**Comment by ilya-zuyev** [comment]
yeah, you're right. changing `kind`

**Comment by vigothehacker** [comment]
Hi , I am facing this exact same error. I have set the /tmp directory to the 1777 mode . Deleting the dir /tmp and then remaking it from the same user makes it work for the current user but fails again when running minikube from a different user.

Please mark https://github.com/kubernetes/minikube/issues/11254 as duplicate for this issue

**Comment by medyagh** [comment]
how about a Lock here?
```
	// make sure log_dir exists if log_file is not also set - the log_dir is mutually exclusive with the log_file option
	// ref: https://github.com/kubernetes/klog/blob/52c62e3b70a9a46101f33ebaf0b100ec55099975/klog.go#L491
	if pflag.Lookup("log_file") != nil && pflag.Lookup("log_file").Value.String() == "" &&
		pflag.Lookup("log_dir") != nil && pflag.Lookup("log_dir").Value.String() != "" {
		if err := os.MkdirAll(pflag.Lookup("log_dir").Value.String(), 0755); err != nil {
			klog.Warningf("unable to create log directory: %v", err)
		}
	}
	```

**Comment by andriyDev** [comment]
Fixed in #11294.
